src/iop/crop.c

```diff
@@ -208,10 +245,13 @@ static void _commit_box(dt_iop_module_t *self,
       p->ch = CLAMPF(p->ch, 0.1f, 1.0f);
     }
   }
+
+  if(aligned) p->aligned = TRUE;
   const gboolean changed =  !feqf(p->cx, old[0], eps)
                         ||  !feqf(p->cy, old[1], eps)
                         ||  !feqf(p->cw, old[2], eps)
-                        ||  !feqf(p->ch, old[3], eps);
+                        ||  !feqf(p->ch, old[3], eps)
+                        || aligned ;
 
```

Comment:
Have you checked whether this causes extranous history entries when merely repeatedly enabling/disabling the module, or even just giving it focus, when `aligned` is true?  We had that problem many years ago with the old crop&rotate module.

Reply:
Yes. We call `_commit_box()` with `aligned==TRUE` only when button is released **and** ` !p->aligned`